CE-521 - Revamp API gateway documentation (#1790)
hashicorp/web-unified-docs: content/consul/v1.22.x/content/docs/north-south/api-gateway/k8s/index.mdx

Refer to [Enable API gateway for Kubernetes](/consul/docs/north-south/api-gateway/k8s/enable) to verify the configuration parameters in your installation.

## Deploy a listener

API gateways have one or more listeners that serve as ingress points for requests to services in a Consul service mesh.

Refer to [Deploy API gateway listeners in Kubernetes](/consul/docs/north-south/api-gateway/k8s/listener) for guidance.

## Define a route

Routes are rule-based configurations that allow external clients to send requests to services in the mesh.

[Define API gateway routes on Kubernetes](/consul/docs/north-south/api-gateway/k8s/route) helps you define a route from your API gateway to a service.

Routes can be tuned to define complex traffic shaping, refer to [Reroute HTTP requests](/consul/docs/north-south/api-gateway/k8s/reroute) to learn hot to define URL rewrite rules for the services in your Kubernetes cluster.

API gateways can also be used to expose services exported by peer datacenters, [Route traffic to peered services](/consul/docs/north-south/api-gateway/k8s/peer) contains guidance on how to expose an exported service using API gateway.

## Guidance

> Refer to the following resources for help setting up and using API gateways:

Reviewer: ```suggestion
The following resources are available to help you use Consul API gateways on Kubernetes.
```

### Tutorials

- [Enable external traffic ingress into Consul service mesh](/consul/tutorials/get-started-kubernetes/kubernetes-gs-ingress)
- [Control access into the service mesh with Consul API gateway](/consul/tutorials/developer-mesh/kubernetes-api-gateway)

### Usage documentation

- [Deploy API gateway listeners to Kubernetes](/consul/docs/north-south/api-gateway/k8s/listener)
- [Deploy API gateway routes to Kubernetes](/consul/docs/north-south/api-gateway/k8s/route)
- [Reroute HTTP requests in Kubernetes](/consul/docs/north-south/api-gateway/k8s/reroute)
- [Route traffic to peered services in Kubernetes](/consul/docs/north-south/api-gateway/k8s/peer)
- [Use JWTs to verify requests to API gateways on Kubernetes](/consul/docs/north-south/api-gateway/secure-traffic/jwt/k8s)

### Reference

> 

Reviewer: ```suggestion
### Reference documentation

To use Consul API gateway on Kubernetes, you must configure the following custom resource definitions (CRDs).

```

| [`GatewayClass`](/consul/docs/reference/k8s/api-gateway/gatewayclass)             | Defines a class of gateway resources used as a template for creating gateways. The default gateway class is `consul` and is suitable for most API gateway implementations.  | 
| [`GatewayClassConfig`](/consul/docs/reference/k8s/api-gateway/gatewayclassconfig) | Describes additional gateway-related configuration parameters for the `GatewayClass` resource. | 
| [`Routes`](/consul/docs/reference/k8s/api-gateway/routes)                         | Specifies paths from the gateway listener to backend services. | 
| [`MeshServices`](/consul/docs/north-south/api-gateway/k8s/peer)                   | Enables routes to reference services in Consul. | 
| [`ServiceIntentions`](/consul/docs/reference/config-entry/service-intentions)     | Specifies traffic communication rules between services in the mesh. Intentions also enforce rules for service-to-service traffic routed through a Consul API gateway. | 

> ### Troubleshooting

Reviewer: ```suggestion
## Troubleshooting
```
